app/models/comment.rb

```diff
@@ -1,4 +1,10 @@
 class Comment < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  def update_counter(current_post_id = post_id)
+    post = Post.find(current_post_id)
+    post.comments_counter = Comment.where(post_id: post.id).count
+    post.save
+  end
```

Comment:
- [ ] Your `update_counter` method for the Comment model gets the post's comments but is not updating(incrementing) the comments counter. Please add the necessary logic to fix this.
For instance, you can try using `post.increment!(:comments_counter)` or `post.comments_counter += 1`

[OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new post has been saved to database, then make the `update_comments_counter` method private

```
  # app/models/comment.rb
  after_save :update_comments_counter

  private

  def update_comments_counter
    # logic here
  end

```

Reply:
And it's the same for this.